homeassistant/components/reolink/light.py

```diff
@@ -0,0 +1,153 @@
+        is_on=lambda api, ch: api.status_led_enabled(ch),
+        turn_on_off=lambda api, ch, value: api.set_status_led(ch, value),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Reolink light entities."""
+    reolink_data: ReolinkData = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        ReolinkLightEntity(reolink_data, channel, entity_description)
+        for entity_description in LIGHT_ENTITIES
+        for channel in reolink_data.host.api.channels
+        if entity_description.supported(reolink_data.host.api, channel)
+    )
+
+
+class ReolinkLightEntity(ReolinkCoordinatorEntity, LightEntity):
+    """Base light entity class for Reolink IP cameras."""
+
+    entity_description: ReolinkLightEntityDescription
+
+    def __init__(
+        self,
+        reolink_data: ReolinkData,
+        channel: int,
+        entity_description: ReolinkLightEntityDescription,
+    ) -> None:
+        """Initialize Reolink light entity."""
+        super().__init__(reolink_data, channel)
+        self.entity_description = entity_description
+
+        self._attr_unique_id = (
+            f"{self._host.unique_id}_{self._channel}_{entity_description.key}"
+        )
+
+        if self.entity_description.set_brightness is None:
+            self._attr_supported_color_modes = {ColorMode.ONOFF}
+            self._attr_color_mode = ColorMode.ONOFF
+        else:
+            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+            self._attr_color_mode = ColorMode.BRIGHTNESS
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if light is on."""
+        return self.entity_description.is_on(self._host.api, self._channel)
+
+    @property
+    def brightness(self) -> int | None:
+        """Return the brightness of this light between 0.255."""
+        if self.entity_description.get_brightness is None:
+            return None
+
+        return round(
+            255
+            * (
+                self.entity_description.get_brightness(self._host.api, self._channel)
+                / 100.0
+            )
+        )
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn light off."""
+        await self.entity_description.turn_on_off(self._host.api, self._channel, False)
+        self.async_write_ha_state()
+
+    async def async_turn_on(self, **kwargs):
+        """Turn light on."""
+        if (
+            brightness := kwargs.get(ATTR_BRIGHTNESS)
+        ) is not None and self.entity_description.set_brightness is not None:
+            brightness_pct = int(brightness / 255.0 * 100)
+            await self.entity_description.set_brightness(
+                self._host.api, self._channel, brightness_pct
+            )
+
+        await self.entity_description.turn_on_off(self._host.api, self._channel, True)
+        self.async_write_ha_state()
```

Comment:
Will the API always be updated directly after the call?

Reply:
Yes, setting properties on the host/camera is always done using the `send_setting` function which has these lines:
https://github.com/starkillerOG/reolink_aio/blob/20d4765b0c664878c6a9a02b7335051d23389021/reolink_aio/api.py#L2845-L2849
```
        if command[:3] == "Set":
            getcmd = command.replace("Set", "Get")
            if wait_before_get > 0:
                await asyncio.sleep(wait_before_get)
            await self.get_state(cmd=getcmd)
```

So it will indeed fetch the state from the API after changing the light status.